pkg/monitor/disruption_known_backends.go

```diff
@@ -0,0 +1,136 @@
+package monitor
+
+import (
+	"context"
+
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/transport"
+)
+
+// this entire file should be a separate package with disruption_***, but we are entanged because the sampler lives in monitor
+// and the things being started by the monitor are coupled into .Start.
+// we also got stuck on writing the disruption backends.  We need a way to track which disruption checks we have started,
+// so we can properly write out "zero"
+
+var (
+	LocatorKubeAPIServerNewConnection         = LocateDisruptionCheck("kube-api", NewConnectionType)
+	LocatorKubeAPIServerReusedConnection      = LocateDisruptionCheck("kube-api", ReusedConnectionType)
+	LocatorOpenshiftAPIServerNewConnection    = LocateDisruptionCheck("openshift-api", NewConnectionType)
+	LocatorOpenshiftAPIServerReusedConnection = LocateDisruptionCheck("openshift-api", ReusedConnectionType)
+	LocatorOAuthAPIServerNewConnection        = LocateDisruptionCheck("oauth-api", NewConnectionType)
+	LocatorOAuthAPIServerReusedConnection     = LocateDisruptionCheck("oauth-api", ReusedConnectionType)
+)
+
+// BackendDisruptionLocatorsToName maps from the locator name used to track disruption to the name used to recognize it in
+// the job aggregator.
+var BackendDisruptionLocatorsToName = map[string]string{
```

Comment:
This is currently being used in the in our `write_job_run_data.go` to be sure we correctly write out "zero disruption" results.

Once we have collapsed on the single sampler implementation, we can always write out a, "sampling backend X" that can be the marker in the monitor.